Sources/UI/Chat/Cells/MessageTableViewCell+Attachments.swift

```diff
@@ -40,96 +38,76 @@ extension MessageTableViewCell {
                 .disposed(by: disposeBag)
         }
         
-        let preview: AttachmentPreviewProtocol
+        let preview: AttachmentPreview
         let isFileAttachment = attachment.type == .file
         
         if isFileAttachment {
-            preview = createAttachmentFilePreview(with: attachment, style: style)
+            preview = FileAttachmentPreview(frame: .zero)
+            preview.setup(attachment: attachment, style: style)
+            preview.snp.makeConstraints { $0.height.equalTo(CGFloat.attachmentFilePreviewHeight).priority(999) }
         } else {
-            preview = createAttachmentPreview(with: attachment,
-                                              style: style,
-                                              imageBackgroundColor: imageBackgroundColor,
-                                              reload: reload)
+            let imagePreview = ImageAttachmentPreview(frame: .zero)
+            imagePreview.setup(attachment: attachment, style: style)
+            imagePreview.forceToReload = reload
+            preview = imagePreview
         }
         
+        preview.index = index
         messageStackView.insertArrangedSubview(preview, at: index)
         attachmentPreviews.append(preview)
         
         // File preview.
-        if isFileAttachment {
-            preview.update(maskImage: backgroundImageForAttachment(at: index)) { _, _ in }
+        guard let imagePreview = preview as? ImageAttachmentPreview else {
+            preview.update { _, _ in }
```

Comment:
Why do we need this empty call before the actual call?

Reply:
The first update is for updating the state of the view.
The second is update the `imageMask`.

Reply:
Hmm, this seems like a code smell. Why do we need to call `update` twice? Can we combine the logic?

Reply:
It was a single method with several logics inside, I separated it and now it's easy to manage.

Reply:
Do you suggest anything? I know it needs time to understand. Ask me.

Reply:
`update(imageMask:)` could be renamed to `apply(imageMask:)`
And in `func update(_ completion: @escaping Completion)`, parameter `completion` could be optional, so we don't need to do `preview.update { _, _ in }`, just `preview.update()`

Reply:
Done.